Sources/SwiftlyCore/ToolchainVerison.swift

```diff
@@ -334,7 +334,7 @@ struct StableReleaseParser: ToolchainSelectorParser {
 ///    - a.b-SNAPSHOT
 struct ReleaseSnapshotParser: ToolchainSelectorParser {
     static let regex: Regex<(Substring, Substring, Substring, Substring?)> =
-        try! Regex("^([0-9]+)\\.([0-9]+)-(?:snapshot|DEVELOPMENT-SNAPSHOT|SNAPSHOT)(?:-([0-9]{4}-[0-9]{2}-[0-9]{2}))?(?:-a)?$")
+        try! Regex("^(?:swift-)?([0-9]+)\\.([0-9]+)-(?:snapshot|DEVELOPMENT-SNAPSHOT|SNAPSHOT)(?:-([0-9]{4}-[0-9]{2}-[0-9]{2}))?(?:-a)?$")
```

Comment:
For ease of implementation and maximum flexibility, I didn't bother creating separate regexes to only allow the prefix when using `DEVELOPMENT-SNAPSHOT` rather than the swiftly canonical name, so selectors like "swift-5.7-snapshot" are allowed.